src/model/ice_client.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2018 Novo Nordisk Foundation Center for Biosustainability, DTU.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+
+import requests
+
+from . import settings
+
+
+logger = logging.getLogger(__name__)
+
+
+class _ICE:
+    """
+    ICE API client. Don't instantiate this class; import the `ice` singleton from the same module.
```

Comment:
Why not create a real singleton then, for example, https://stackoverflow.com/a/6798042

Reply:
That's indeed a better implementation of the pattern, thanks :+1: